### PR TITLE
[CID-883] - Modify 'Create IDP' to support metadata.xml

### PR DESCRIPTION
**Pull request by kidster**
**Quality audit reminder**

Doc team: While logged in using your O365 account, complete the quality checklist at http://bit.ly/2hwvBKX. Use the checklist for significant doc changes or additions submitted by you or a non-doc staff contributor.

### CODE REVIEW

File: api-docs/api-reference/identity-provider-operations.rst [hunk 1]
```diff
@@ -0,0 +1,28 @@
+.. _identity-provider-operations:
+
+=================
+Identity provider
+=================
+
+An Identity Provider (IDP) is required in order to federate with Identity.
+Services are provided to manage Identity Providers (IDPs) within Identity.
+
+**Access Restrictions**
```
Comment (cyrichardson) [inline api-docs/api-reference/identity-provider-operations.rst]: Change Restrictions to restrictions

File: api-docs/api-reference/methods/identity-provider/post-create-identity-provider-with-metadata-v2.0.rst [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+
+.. code::
+
+   POST /v2.0/RAX-AUTH/federation/identity-providers
+
+Create a new Identity Provider using XML metadata.
+
+.. note::
+
+   - Creating IDP using metadata will auto assign the IDP's name to at most 29
```
Comment (cyrichardson) [inline api-docs/api-reference/methods/identity-provider/post-create-identity-provider-with-metadata-v2.0.rst]: Creating IDP using metadata auto assigns

File: api-docs/api-reference/methods/identity-provider/post-create-identity-provider-with-metadata-v2.0.rst [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+   - If IDP's name already exists, a digit is added at the end of the
+     name until a unique name is found. Ex: example_2
+   - This resource describes a single deployment using `EntityDescriptor`.
+
+This table shows the possible response codes for this operation:
+
+.. csv-table::
+   :header: Response Code, Name, Description
+   :widths: auto
+
```
Comment (cyrichardson) [inline api-docs/api-reference/methods/identity-provider/post-create-identity-provider-with-metadata-v2.0.rst]: Remove " throughout the table.

File: api-docs/api-reference/methods/identity-provider/post-create-identity-provider-with-metadata-v2.0.rst [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+   X-Auth-Token, String *(Required)*, A valid authentication token.
+
+This table shows the body parameters for the request:
+
+.. csv-table::
+   :header: Name, Type, Description
+   :widths: auto
+
+    **EntityDescriptor**, Object *(Required)*, Describes a system entity such as an Identity Provider.
+    EntityDescriptor.\ **entityID**, String *(Required)*, Issuer for IDP.
```
Comment (cyrichardson) [inline api-docs/api-reference/methods/identity-provider/post-create-identity-provider-with-metadata-v2.0.rst]: The issuer

File: api-docs/api-reference/methods/identity-provider/post-create-identity-provider-with-metadata-v2.0.rst [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+    EntityDescriptor.\ **IDPSSODescriptor**, Object *(Required)*, An IDP role.
+    EntityDescriptor.IDPSSODescriptor.\ **protocolSupportEnumeration**, String *(Required)*, Represent general classes of protocol support for the role in question.
```
Comment (cyrichardson) [inline api-docs/api-reference/methods/identity-provider/post-create-identity-provider-with-metadata-v2.0.rst]: Represents

File: api-docs/api-reference/methods/identity-provider/post-create-identity-provider-with-metadata-v2.0.rst [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+    EntityDescriptor.IDPSSODescriptor.\ **SingleSignOnService**, Object *(Required)*, Describes a protocol binding endpoint.
+    EntityDescriptor.IDPSSODescriptor.SingleSignOnService.\ **Binding**, String *(Required)*, Describes a protocol binding. Only `HTTP-Redirect` is currently supported.
+    EntityDescriptor.IDPSSODescriptor.SingleSignOnService.\ **Location**, String *(Required)*, Describes the authentication url.
+    EntityDescriptor.IDPSSODescriptor.\ **KeyDescriptor**, Object *(Optional)*, Associate one or more public keys with the system being defined.
```
Comment (cyrichardson) [inline api-docs/api-reference/methods/identity-provider/post-create-identity-provider-with-metadata-v2.0.rst]: Associates

File: api-docs/api-reference/methods/identity-provider/post-create-identity-provider-with-metadata-v2.0.rst [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+    EntityDescriptor.IDPSSODescriptor.KeyDescriptor.\ **KeyInfo**, Object *(Optional)*, Element describing keys.
```
Comment (cyrichardson) [inline api-docs/api-reference/methods/identity-provider/post-create-identity-provider-with-metadata-v2.0.rst]: The element

File: api-docs/api-reference/methods/identity-provider/post-create-identity-provider-with-metadata-v2.0.rst [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+Response
+--------
+
+This table shows the header parameters for the response:
+
+.. csv-table::
+    :header: Name, Type, Description
+    :widths: auto
+
+    Location, String *(Required)*, Location to the URI of the newly created IDP
```
Comment (cyrichardson) [inline api-docs/api-reference/methods/identity-provider/post-create-identity-provider-with-metadata-v2.0.rst]: The location to the URI of the newly created IDP.

File: api-docs/api-reference/methods/identity-provider/post-create-identity-provider-with-metadata-v2.0.rst (hunk 3 above)
Comment (kidster) [inline api-docs/api-reference/methods/identity-provider/post-create-identity-provider-with-metadata-v2.0.rst]: The `"` are needed for line continuation.

Comment (cyrichardson) [inline api-docs/api-reference/methods/identity-provider/post-create-identity-provider-with-metadata-v2.0.rst]: Thanks for letting me know!

**Pull request merged**